src/utils/ProfileProvider.tsx

```diff
@@ -19,16 +22,23 @@ interface ProfileContextType {
   profileData: Profile | null;
   languages: ProfileLanguage[];
   roles: ProfileRole[];
-  updateProfile: (newProfileData: Partial<Profile>) => void;
-  setLanguages: (languages: ProfileLanguage[]) => void;
-  setRoles: (roles: ProfileRole[]) => void;
+  userId: UUID | undefined;
+  updateProfile: (newProfileData: Partial<Profile>) => Promise<void>;
+  setLanguages: (languages: ProfileLanguage[]) => Promise<void>;
+  setRoles: (roles: ProfileRole[]) => Promise<void>;
+  createNewProfile: (
+    profile: Profile,
+    languages: ProfileLanguage[],
+    roles: ProfileRole[],
+  ) => Promise<void>;
 }
 
 export const ProfileContext = createContext<ProfileContextType | undefined>(
   undefined,
 );
 
 export default function ProfileProvider({ children }: { children: ReactNode }) {
+  const [userId, setUserId] = useState<UUID | undefined>(undefined);
```

Comment:
nice!! this is v helpful - helps us distinguish between the un-authenticated vs. authenticated + un-onboarded states on the `/cases` page

to [this question](https://www.notion.so/calblueprint/council-3f1376dc1ec640adb82748cd7e716e6f?pvs=4#284b0f1ff79a4157a2f8fdaf71a23d01) that you asked btw, i wonder if just creating an auth context with this useState (and also one for the session) as well as all of the auth functions would be cleanest....thereby separating that from the `ProfileContext` entirely

(see [this](https://github.com/calblueprint/girls-write-now/blob/main/src/utils/AuthContext.tsx) from gwn)

Reply:
Hmm, I think we should discuss that on tues! Looking at gwn's auth context, it looks like the primary benefit we get is making sure the authenticated user object is kept in sync. Not too sure how our auth currently looks like, but I think we're checking credentials via Supabase every time? (i.e. we don't keep track of the user object) One may be better than the other, will need more research.